File: Languages/JavaScript/BubbleSort.js
#!/usr/bin/env node

var LOGGING = false;

function bubbleSort(arr) {
    let len = arr.length;
    for (let i = 0; i < len; i++) {
        for (let j = 0; j < len - 1 - i; j++) {
            if (arr[j] > arr[j + 1]) {
                let temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
    return arr;
}

function main() {
    let arr = [10650,56341,77676,47158,34999,22009,19680,41870,31090,30668]
    bubbleSort(arr);
    if (LOGGING) {
        console.log("Sorted Array: " + arr);
    } else {
        console.log();
    }
}

main();

File: Languages/Ruby/BubbleSort.rb
#!/usr/bin/env ruby

# Source: ChatGPT

$LOGGING = false

def bubble_sort(arr)
  n = arr.length

  # Traverse through all array elements
  (0..n-1).each do |i|
    # Last i elements are already in place
    (0..n-i-2).each do |j|
      # Traverse the array from 0 to n-i-1
      # Swap if the element found is greater than the next element
      if arr[j] > arr[j+1]
        temp = arr[j]
        arr[j] = arr[j+1]
        arr[j+1] = temp
      end
    end
  end
  arr
end


def main()
  arr = [10650,56341,77676,47158,34999,22009,19680,41870,31090,30668]
  bubble_sort(arr)
  if $LOGGING
    puts "Sorted array: #{arr}"
  else
    puts ""
  end
end

main()